### ISSUE TITLE
Linked TF crash when loading TF preset

### ISSUE THREAD

**Issue by martinfalk** [inviwo/inviwo#1053]
Loading a TF preset for a linked TF property results in a read access violation. The LinkEvaluator is triggered while the deserializer is still working on extracting the TF primitives.

https://github.com/inviwo/inviwo/blob/5617847717c9eef9cf8870001bc7277f07208002/src/core/datastructures/tfprimitiveset.cpp#L341-L352

To recreate:
1. load boron.inv
2. add `Color Scale Legend` before the background processor
3. link the TFs of Volume Raycaster and the color legend
4. right click on TF, load any `*.itf` TF preset

**Comment by martinfalk** [comment]
Call Stack
```
 	inviwo-core.dll!inviwo::operator==(const inviwo::TFPrimitiveData & lhs, const inviwo::TFPrimitiveData & rhs) Line 129	C++
 	inviwo-core.dll!inviwo::operator!=(const inviwo::TFPrimitive & lhs, const inviwo::TFPrimitive & rhs) Line 150	C++
 	inviwo-core.dll!inviwo::TFPrimitive::operator=(const inviwo::TFPrimitive & rhs) Line 54	C++
 	inviwo-core.dll!inviwo::TFPrimitiveSet::operator=(const inviwo::TFPrimitiveSet & rhs) Line 79	C++
 	inviwo-core.dll!inviwo::TransferFunction::operator=(const inviwo::TransferFunction & rhs) Line 89	C++
 	[Inline Frame] inviwo-core.dll!inviwo::ValueWrapper<inviwo::TransferFunction>::update(const inviwo::TransferFunction &) Line 114	C++
 	inviwo-core.dll!inviwo::TransferFunctionProperty::set(const inviwo::TransferFunction & value) Line 222	C++
 	inviwo-core.dll!inviwo::CompositeProperty::set(const inviwo::CompositeProperty * src) Line 71	C++
 	inviwo-core.dll!inviwo::LinkEvaluator::evaluateLinksFromProperty(inviwo::Property * modifiedProperty) Line 172	C++
 	inviwo-core.dll!inviwo::ProcessorNetwork::onAboutPropertyChange(inviwo::Property * modifiedProperty) Line 375	C++
 	[Inline Frame] inviwo-core.dll!inviwo::ProcessorObservable::notifyObserversAboutPropertyChange::__l2::<lambda_7b9ec18cae532aa6e14609f4abd3e1ea>::operator()(inviwo::ProcessorObserver * o) Line 116	C++
 	inviwo-core.dll!inviwo::Observable<inviwo::ProcessorObserver>::forEachObserver<<lambda_7b9ec18cae532aa6e14609f4abd3e1ea>>(inviwo::ProcessorObservable::notifyObserversAboutPropertyChange::__l2::<lambda_7b9ec18cae532aa6e14609f4abd3e1ea> callback) Line 303	C++
 	[Inline Frame] inviwo-core.dll!inviwo::ProcessorObservable::notifyObserversAboutPropertyChange(inviwo::Property * p) Line 116	C++
 	inviwo-core.dll!inviwo::Property::propertyModified() Line 181	C++
 	inviwo-core.dll!inviwo::Property::propertyModified() Line 186	C++
 	[Inline Frame] inviwo-core.dll!inviwo::TFPrimitiveSetObservable::notifyTFPrimitiveChanged::__l2::<lambda_79c19e9d71e9e9968d29b43da479f9bf>::operator()(inviwo::TFPrimitiveSetObserver * o) Line 58	C++
 	inviwo-core.dll!inviwo::Observable<inviwo::TFPrimitiveSetObserver>::forEachObserver<<lambda_79c19e9d71e9e9968d29b43da479f9bf>>(inviwo::TFPrimitiveSetObservable::notifyTFPrimitiveChanged::__l2::<lambda_79c19e9d71e9e9968d29b43da479f9bf> callback) Line 303	C++
 	[Inline Frame] inviwo-core.dll!inviwo::TFPrimitive::notifyTFPrimitiveObservers::__l2::<lambda_4d474c6c1ff44752dc49640873e66099>::operator()(inviwo::TFPrimitiveObserver * o) Line 112	C++
 	inviwo-core.dll!inviwo::Observable<inviwo::TFPrimitiveObserver>::forEachObserver<<lambda_4d474c6c1ff44752dc49640873e66099>>(inviwo::TFPrimitive::notifyTFPrimitiveObservers::__l2::<lambda_4d474c6c1ff44752dc49640873e66099> callback) Line 303	C++
 	inviwo-core.dll!inviwo::TFPrimitive::deserialize(inviwo::Deserializer & d) Line 126	C++
 	inviwo-core.dll!inviwo::Deserializer::deserialize(const std::string & key, inviwo::Serializable & sObj) Line 71	C++
 	inviwo-core.dll!inviwo::Deserializer::deserialize<inviwo::TFPrimitive>(const std::string & key, inviwo::TFPrimitive * & data) Line 1131	C++
 	inviwo-core.dll!inviwo::Deserializer::deserialize<inviwo::TFPrimitive>(const std::string & key, std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>> & data) Line 1153	C++
 	inviwo-core.dll!inviwo::ContainerWrapper<std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>>,std::string>::deserialize(inviwo::Deserializer & d, ticpp::Element * node, unsigned __int64 ind) Line 346	C++
 	[External Code]	
 	[Inline Frame] inviwo-core.dll!std::_Func_class<void,ticpp::Element *>::operator()(ticpp::Element * <_Args_0>) Line 951	C++
 	inviwo-core.dll!inviwo::detail::forEachChild(ticpp::Element * node, const std::string & key, std::function<void __cdecl(ticpp::Element *)> func) Line 145	C++
 	inviwo-core.dll!inviwo::Deserializer::deserialize<std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>>,std::string>(const std::string & key, inviwo::ContainerWrapper<std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>>,std::string> & container) Line 1162	C++
 	inviwo-core.dll!inviwo::util::IndexedDeserializer<std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>>>::operator()<std::vector<std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>>,std::allocator<std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>>>>>(inviwo::Deserializer & d, std::vector<std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>>,std::allocator<std::unique_ptr<inviwo::TFPrimitive,std::default_delete<inviwo::TFPrimitive>>>> & container) Line 431	C++
>	inviwo-core.dll!inviwo::TFPrimitiveSet::deserialize(inviwo::Deserializer & d) Line 341	C++
 	inviwo-core.dll!inviwo::TransferFunction::deserialize(inviwo::Deserializer & d) Line 135	C++
 	inviwo-core.dll!inviwo::TransferFunction::load(const std::string & filename, const inviwo::FileExtension & ext) Line 187	C++
 	[Inline Frame] inviwo-module-qtwidgets.dll!inviwo::util::addTFPresetsMenu::__l8::void <lambda>(const std::string &)::__l13::<lambda_03f13214bcd231e69878f88dd6d2beb7>::operator()() Line 135	C++
```

**Comment by petersteneteg** [comment]
fix in #1066
